packages/render-fidelity-tools/src/components/renderers/babylon-viewer.ts

```diff
@@ -0,0 +1,151 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '@babylonjs/loaders/glTF';
+import {ArcRotateCamera, Engine, HDRCubeTexture, Scene, SceneLoader, Vector3} from '@babylonjs/core';
+import {css, customElement, html, LitElement, property} from 'lit-element';
+
+import {ScenarioConfig} from '../../common.js';
+
+// const IS_BINARY_RE = /\.glb$/;
+
+const $initialize = Symbol('initialize');
+const $updateScenario = Symbol('scenario');
+const $updateSize = Symbol('updateSize');
+const $render = Symbol('render');
+const $canvas = Symbol('canvas');
+const $engine = Symbol('engine');
+const $scene = Symbol('scene');
+const $camera = Symbol('camera');
+const $hdrTexture = Symbol('hdrTexture');
+
+
+@customElement('babylon-viewer')
+export class BabylonViewer extends LitElement {
+  @property({type: Object}) scenario: ScenarioConfig|null = null;
+  private[$canvas]: HTMLCanvasElement|null;
+  private[$engine]: Engine;
+  private[$scene]: Scene;
+  private[$camera]: ArcRotateCamera;
+  private[$hdrTexture]: HDRCubeTexture;
+
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+  }
+
+  disconnectedCallback() {
+  }
+
+  updated(changedProperties: Map<string, any>) {
+    super.updated(changedProperties);
+    this[$updateSize]();
+
+    if (changedProperties.has('scenario') && this.scenario != null) {
+      this[$updateScenario](this.scenario);
+    }
+  }
+
+  static get styles() {
+    return css`
+:host {
+ display: block;
+}
+`;
+  }
+
+  render() {
+    this[$render]();
+    return html`<canvas id="canvas"></canvas>`;
+  }
+
+
+  private[$initialize]() {
+    this[$canvas] = this.shadowRoot!.querySelector('canvas');
+    this[$engine] = new Engine(this[$canvas], true);
+    this[$scene] = new Scene(this[$engine]);
+  }
+
+
+  private async[$updateScenario](scenario: ScenarioConfig) {
+    if (this[$scene] != null) {
+      this[$scene].dispose();
+    }
+
+    this[$initialize]();
+    this[$updateSize]();
+
+    // create camera
+    const {orbit, target} = scenario;
+    const alpha = orbit.phi * Math.PI / 180;
+    const beta = orbit.phi * Math.PI / 180;
+    this[$camera] = new ArcRotateCamera(
+        'Camera',
+        alpha,
+        beta,
+        orbit.radius,
+        new Vector3(target.x, target.y, target.z),
+        this[$scene]);
+    this[$camera].attachControl(this[$canvas]!, true);
+
+    // load model
+    const lastSlashIndex = scenario.model.lastIndexOf('/');
+    const modelRootPath = scenario.model.substring(0, lastSlashIndex + 1);
+    const modelFileName =
+        scenario.model.substring(lastSlashIndex + 1, scenario.model.length);
+
+    await new Promise((resolve) => {
+      SceneLoader.Append(modelRootPath, modelFileName, this[$scene], () => {
```

Comment:
`SceneLoader.AppendAsync` returns a promise, so I think instead you could do:

```
await SceneLoader.AppendAsync(...)
```

That being said, it's unclear to me what the difference is between `Append` and `AppendAsync`.. maybe `Append` is already synchronous?

Reply:
i'm think you are right. I will replace that with the AppendAsync function. Thanks for the comment!